src/core/server/services/giphy/giphy.ts

```diff
@@ -0,0 +1,123 @@
+import { URL } from "url";
+
+import { GIPHY_FETCH, GIPHY_SEARCH } from "coral-common/constants";
+import { LanguageCode } from "coral-common/helpers";
+import { createFetch } from "coral-server/services/fetch";
+
+import { GQLGIPHY_RATING } from "coral-server/graph/schema/__generated__/types";
+
+const fetch = createFetch({ name: "giphy" });
+
+type GiphyLanguage = "en" | "es" | "fr" | "de" | "pt";
+
+export function ratingIsAllowed(rating: string, maxRating: GQLGIPHY_RATING) {
+  const ratingsOrder = [
+    GQLGIPHY_RATING.G,
+    GQLGIPHY_RATING.PG,
+    GQLGIPHY_RATING.PG13,
+    GQLGIPHY_RATING.R,
+  ];
+  const compareRating = rating.toUpperCase() as GQLGIPHY_RATING;
+  return (
+    ratingsOrder.includes(compareRating) &&
+    ratingsOrder.indexOf(compareRating) <= ratingsOrder.indexOf(maxRating)
+  );
+}
+
+/**
+ * convertLanguage returns the language code for the related Perspective API
+ * model in the ISO 631-1 format.
+ *
+ * @param locale the language on the tenant in the BCP 47 format.
+ */
+function convertLanguage(locale: LanguageCode): GiphyLanguage {
+  switch (locale) {
+    case "en-US":
+      return "en";
+    case "es":
+      return "es";
+    case "fr-FR":
+      return "fr";
+    case "de":
+      return "de";
+    case "pt-BR":
+      return "pt";
+    default:
+      return "en";
+  }
+}
+
+export async function searchGiphy(
+  query: string,
+  offset: string,
+  rating: GQLGIPHY_RATING,
+  apiKey: string,
+  locale: LanguageCode
+) {
+  const language = convertLanguage(locale);
+  const url = new URL(GIPHY_SEARCH);
+  url.searchParams.set("api_key", apiKey);
+  url.searchParams.set("limit", "10");
+  url.searchParams.set("lang", language);
+  url.searchParams.set("offset", offset);
+  url.searchParams.set("rating", rating);
+  url.searchParams.set("q", query);
+
+  try {
+    const res = await fetch(url.toString());
+
+    if (!res.ok) {
+      return {
+        ok: res.ok,
+        status: res.status,
+        data: null,
+      };
+    }
+
+    // Parse the JSON body and send back the result!
+    const data = await res.json();
```

Comment:
Should set a specific type for the returned data and validate the shape with `Joi.validate`.